### PR TITLE
Import Chain from Backup

### PR DESCRIPTION
**Pull request by qinwei-gong**

### CODE REVIEW

File: snapshot/chain_correction_export.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+		if _, ok := t.(*types.DepositStakeTx); ok {
+			continue
+		}
+		if _, ok := t.(*types.WithdrawStakeTx); ok {
+			continue
+		}
+
+		hash := crypto.Keccak256Hash(tx).Hex()
+		found := false
+		for _, exclusion := range exclusionTxs {
```
Comment (jieyilong) [inline snapshot/chain_correction_export.go]: Can change exclusionTxs to a map? psudo code

if !exclusionTxMap.find(hash) {
   results = append(results, tx)
}

File: snapshot/chain_correction_export.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+	bhStack := make(BHStack, 0)
+	bhStackRev := make(BHStack, 0)
+
+	for {
+		block.Txs = ExcludeTxs(block.Txs, exclusionTxs, chain)
+		block.TxHash = core.CalculateRootHash(block.Txs)
+		bh := block.UpdateHash()
+		bhStack = bhStack.push(bh)
+		chain.SaveBlock(block)
```
Comment (jieyilong) [inline snapshot/chain_correction_export.go]: do we want to save the block to the chain or better store it in a temporary data structure?

File: snapshot/chain_correction_export.go (hunk 1 above)
Comment (qinwei-gong) [reply snapshot/chain_correction_export.go]: yah, it makes sense, esp. when there's a lot of exclusionTxs

File: snapshot/chain_correction_export.go (hunk 2 above)
Comment (qinwei-gong) [reply snapshot/chain_correction_export.go]: i was just afraid there might be too many blocks to save and it'd consume a lot of memory. the old chain will be wiped out anyway?

**Pull request merged**